packages/mermaid/src/rendering-util/selectSvgElement.ts

```diff
@@ -0,0 +1,24 @@
+import { select } from 'd3';
+import { getConfig } from '../config.js';
+import type { HTML, SVG } from '../diagram-api/types.js';
+
+/**
+ * Selects the SVG element using {@link id}.
+ *
+ * @param id - The diagram ID.
+ * @returns The selected {@link SVG} element using {@link id}.
+ */
+export const selectSvgElement = (id: string): SVG => {
+  const { securityLevel } = getConfig();
+  // handle root and document for when rendering in sandbox mode
+  let doc: Document = document;
+  if (securityLevel === 'sandbox') {
+    const sandboxElement: HTML = select(`#i${id}`);
+    doc = sandboxElement.node()?.contentDocument ?? doc;
+  }
+  const root: HTML =
+    securityLevel === 'sandbox' ? select(doc.body as HTMLIFrameElement) : select('body');
+
+  const svg: SVG = root.select(`#${id}`);
+  return svg;
```

Comment:
@sidharthv96 It would be better use this function instead of repeating it in each diagram renderer.

Reply:
Far better